lib/travis/build/addons/phantomjs.rb

```diff
@@ -0,0 +1,72 @@
+require 'shellwords'
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class PhantomJs < Base
+        SUPER_USER_SAFE = true
+
+        attr_reader :version
+
+        def after_prepare
+          sh.fold 'install_phantomjs' do
+            sanitize(raw_version)
+
+            unless version
+              sh.echo "Invalid version '#{raw_version}' given. Only 1.9.8 and 2.0.0 are supported at the moment", ansi: :red
+              return
+            end
+
+            sh.echo "Installing PhantomJS #{version}", ansi: :yellow
+
+            sh.if "$(uname) = 'Linux'" do
+              sh.if "$(lsb_release -c) = 'precise'" do
+                if version == '2.0.0'
+                  sh.mkdir "/tmp/travis-phantomjs"
+                  sh.cmd "wget https://s3.amazonaws.com/travis-phantomjs/phantomjs-2.0.0-ubuntu-12.04.tar.bz2 -O /tmp/travis-phantomjs/phantomjs-2.0.0-ubuntu-12.04.tar.bz2", echo: true, timing: true, retry: true
+                  sh.cmd "tar -xvf /tmp/travis-phantomjs/phantomjs-2.0.0-ubuntu-12.04.tar.bz2 -C /tmp/travis-phantomjs", echo: true, timing: true, retry: false
+                  sh.export "PATH", "/tmp/travis-phantomjs:$PATH"
+                else
+                  # no need to do anything for 1.9.8 at the moment, it's the
+                  # default on precise
+                end
+              end
+
+              sh.else do
+                if version == '1.9.8'
```

Comment:
Since we have 1.9.8 pre-installed currently, there is no need to do anything on this logical branch.